packages/node/src/transports/base.ts

```diff
@@ -47,30 +57,28 @@ export abstract class BaseTransport implements Transport {
   }
 
   /** Returns a build request option object used by request */
-  protected _getRequestOptions(): http.RequestOptions | https.RequestOptions {
+  protected _getRequestOptions(url: url.URL): http.RequestOptions | https.RequestOptions {
```

Comment:
@kamilogorek I made this take only `url.URL` instead of `string | url.URL`, because we don't really need that flexibility -- and we can add it later if we ever need it. Less lines to maintain.